module-2/backend/week_two.md

```diff
@@ -1,37 +1,93 @@
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+ * A foreign key on a table like payload which had request_type_id would link to a separate request_type table. In that request type table it would contain a primary key of http_verb. If you expect to get a lot of repeated data often times it is best to create a foreign key that bring you to a new table.
+
 10. What are the parts of an HTTP response?
-11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+ * hmm? response the serves gives back to the client
 
+11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+ * On the testing side of things you can create helper methods that you call so you don't have to type the same thing over and over in each test. Also on the views part you can put info in the body so you don't have to type it out in each view.
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+ * find(num) - give you the instance of what ever number you type in.
+ * find_by(name: Anthony) - will return the first name: anthony it finds
+ * where(name: Anthony) - will return all the anthony's it finds
+ * group(:name) - is similar to group_by in ruby
+ * pluck(:name) - give you all the names in an array
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+ * hmm not exactly sure what is meant by rake tasks. Like rake db:migrate, rake db:create, rake db:rollback?
+
 4. What can you expect from a group as you begin working together? As you continue working together?
+ * Super important to DTR and make sure everyone is on the same page.
+
 5. What two columns does `t.timestamps null: false` create in our database?
+ * created_at and updated_at
+
 6. What cURL flag can you use to send a `POST` request?
+ * ??
 7. What case does JSON (and JavaScript) use for multi-word variables?
+ * camelcase
+
 8. What case does Ruby use for multi-word variables?
+ * snakecase
+
 9. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+
+ * School will have_many teachers and a teacher will belong to a school.
+
 10. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+
+![Alt text](schools.png)  
+
 11. Give an example of when you might want to store information besides ids on a join table.
+ * You might want to store a primary on a join table if those tables that are using the join table could share the same information
+
 12. Describe and diagram the relationship between patients and doctors.
+ * In most cased a Doctor has many patients and a patient belongs_to a Doctor. You could also make the case that it will be a many to many relationship as a patient could have many doctors as well.
```

Comment:
I like how you thought this through. These can be a little bit ambiguous, but you've covered your bases.